Test_current_charge_range.cpp

```diff
@@ -0,0 +1,23 @@
+#define CATCH_CONFIG_MAIN  // This tells Catch to provide a main() - only do this in one cpp file
+
+#include "current_charge_range.hpp"
+#include "test/catch.hpp"
+#include <iostream>
+#include <sstream>
+
+int current_charge[] = { 1, 2, 5, 4, 10, 2, 4, 4, 2 };
+
+TEST_CASE("Show range for current samples")
```

Comment:
This is very minimal test case. Verifying only one expectation. What happens when the sample is empty set or repeated values or invalid range is provided?

Setting the expectations as test cases before implementation is very important !